Fix playback of "music choice" cable channels in USA (video frames every 6s)

tmm1 opened the pull request:
Playing back https://s3.amazonaws.com/tmm1/music.ts (and especially when playing that channel live from my network TV tuner), there are a ton of issues:

- [x] "Invalid video timestamp" warnings shown
- [x] player does not start until first video frame, introducing up to 6s delay in playback
- [x] "Audio/Video desynchronisation detected" warning is shown
- [x] player tries to correct for A/V sync and delays video frames incorrectly
- [x] demuxer underrun triggers after each video frame is shown causing repeated "buffering" (when streaming live)
- [x] playback-time is only incremented every 6s

tmm1 commented:
> playback-time is only incremented every 6s

Forcing `handle_playback_time()` to use audio_pts fixed this on the terminal. However, when using libmpv it doesn't help because EVENT_TICK is only fired on video frame display (and on seeks).

wm4 commented:
Nothing is actually supposed to use EVENT_TICK. Using the video timestamp (at least in the case of normal video) is fully intentional, because you can't get exact video frame PTS with audio PTS.

Playing the old "let's adjust this tuned constant to suit the test file I'm looking at" won't lead to anything good, because that's what happens every time the constant is changed, and was also the reason for introducing it.

tmm1 commented:
With ffmpeg master, such video streams are marked as AV_DISPOSITION_STILL_IMAGE (https://github.com/FFmpeg/FFmpeg/commit/5dfeb7f0811c0b64d68e80296067fa8fed57beee), so it should be easier to detect and deal with these now without resorting to this random video timestamp threshold constant.

wm4 commented:
Well, you still need to decide _when_ to show a new frame. I guess you could somehow bind it to the audio PTS.

tmm1 commented:
> Playing the old "let's adjust this tuned constant to suit the test file I'm looking at" won't lead to anything good, because that's what happens every time the constant is changed, and was also the reason for introducing it.

I've removed the special tolerance constant, and updated this PR to special case playback of sparse video streams that are tagged with AV_DISPOSITION_STILL_IMAGE.

tmm1 commented on player/video.c:
```diff
@@ -363,11 +364,15 @@ static void handle_new_frame(struct MPContext *mpctx)
     double pts = mpctx->next_frames[0]->pts;
     if (mpctx->video_pts != MP_NOPTS_VALUE) {
         frame_time = pts - mpctx->video_pts;
-        double tolerance = mpctx->demuxer->ts_resets_possible ? 5 : 1e4;
```
Maybe it's better to leave this in place. It was added in 5f97bd4223 for #3027
